src/parser/shaman/elemental/modules/azerite/LavaShock.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+import { formatNumber } from 'common/format';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'interface/others/TraitStatisticBox';
+import { calculateAzeriteEffects } from 'common/stats';
+
+
+class LavaShock extends Analyzer {
+  procs = 0;
+  damageGained = 0;
+  traitBonus = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.LAVA_SHOCK.id);
+
+    this.traitBonus = this.selectedCombatant.traitsBySpellId[SPELLS.LAVA_SHOCK.id]
+      .reduce((sum, rank) => sum + calculateAzeriteEffects(SPELLS.LAVA_SHOCK.id, rank)[0], 0);
+  }
+
+  on_byPlayer_damage(event) {
+    if (event.ability.guid !== SPELLS.EARTH_SHOCK.id) {
+      return;
+    }
+    const buff = this.selectedCombatant.getBuff(SPELLS.LAVA_SHOCK_BUFF.id);
+    if (buff === undefined) {
+      return;
+    }
+    this.damageGained += buff.stacks * this.traitBonus;
```

Comment:
Per the discussion on discord, this is probably not accurate in terms of how much damage it actually provides. A warlock PR is going to put through a helper method soon to more accurately calculate it.

Reply:
That warlock PR should be already merged so it's now available in `parser/core/calculateBonusAzeriteDamage.js`